volumina/utility/cache.py

```diff
@@ -0,0 +1,119 @@
+import abc
+import sys
+import threading
+from collections import OrderedDict
+from typing import NamedTuple, Dict, Any, Type, TypeVar, Callable
+
+import numpy as np
+
+
+T = TypeVar("T")
+_256_MB = 256 * 1024 * 1024
+
+
+class CacheABC(abc.ABC):
+    @abc.abstractmethod
+    def get(self, key: str, default=None) -> Any:
+        ...
+
+    @abc.abstractmethod
+    def set(self, key: str, value: Any) -> Any:
+        ...
+
+    @abc.abstractmethod
+    def delete(self, key: str) -> None:
+        ...
+
+    @abc.abstractmethod
+    def touch(self, key: str) -> None:
+        ...
+
+    @abc.abstractmethod
+    def __contains__(self, key: str) -> bool:
+        ...
+
+
+class KVCache(CacheABC):
+    _MISSING = object()
+    _cachable_types: Dict[T, Callable[[T], int]] = {}
+
+    class _Entry(NamedTuple):
+        obj: Any
+        size: int  # bytes
+
+    def __init__(self, mem_limit=_256_MB):
+        self._cache = OrderedDict()
+        self._mem_limit = mem_limit
+        self._mem = 0
+        self._lock = threading.RLock()
+
+    def get(self, key: str, default=None) -> Any:
+        with self._lock:
+            entry = self._cache.get(key, self._MISSING)
+            self._cache.move_to_end(key)
+
+        if entry is not self._MISSING:
+            return entry.obj
+        else:
+            return default
+
+    def delete(self, key: str) -> None:
+        with self._lock:
+            self._cache.pop(key, None)
+
+    def keys(self):
```

Comment:
```suggestion
    def keys(self) -> List:
```
with the appropriate import